util/coverage/functest_coverage.py

```diff
@@ -0,0 +1,101 @@
+#!/usr/bin/env python3
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+
+import logging
+from pathlib import Path
+from pprint import pformat
+from typing import List
+
+from device_profile_data import extract_profile_data
+from common import (
+    LLD_TARGET,
+    BazelTestType,
+    CoverageParams,
+    run,
+)
+
+# Commands that this script uses
+LLVM_PROFDATA = "llvm-profdata"
+
+
+def handle_libs(device_libs_all: List[str]) -> List[str]:
```

Comment:
nit: [should these  `handle_*` names be more descriptive](https://google.github.io/styleguide/pyguide.html#316-naming)? maybe `filer_device_libs`? Or are they intended to serve as examples and be overridden/re-implemented for different use cases?

Reply:
@dmcardle also raised a similar [point](https://github.com/lowRISC/opentitan/pull/16947#discussion_r1061765450) for the approach in the previous PR. To be clear: these are callbacks in form (because of the way the initial script was implemented) but specializations (or variants) of a template method in spirit (as in the [template pattern](https://en.wikipedia.org/wiki/Template_method_pattern)). I'd like to finish the tasks in #16761 and see how much of this can be moved to bazel before investing in more significant changes.

Reply:
ah got it. makes sense, just wanted to confirm it was intentional hehe. iterate away